src/Fabulous.AST.Tests/TypeDefinitions/Record.fs

```diff
@@ -0,0 +1,76 @@
+﻿namespace Fabulous.AST.Tests.TypeDefinitions
+
+open FSharp.Compiler.Text
+open Fabulous.AST.Tests
+open Fantomas.Core.SyntaxOak
+open NUnit.Framework
+
+open Fabulous.AST
+open type Ast
+
+module Record =
+    [<Test>]
+    let ``Produces a record`` () =
+
+        AnonymousModule() {
+            Record("Colors") {
+                Field("Red", Type.FromString "int")
+                Field("Green", Type.FromString "int")
+                Field("Blue", Type.FromString "int")
+            }
+        }
+        |> produces
+            """
+
+type Colors = { Red: int; Green: int; Blue: int }
+
+"""
+
+    [<Test>]
+    let ``Produces a record using EscapeHatch`` () =
+        let customField =
+            FieldNode(
+                Some(XmlDocNode([| "/// Super cool doc bro" |], Range.Zero)),
+                None,
+                None,
+                false,
+                None,
+                Some(SingleTextNode("Green", Range.Zero)),
+                Type.FromString "int",
+                Range.Zero
+            )
+
+        AnonymousModule() {
+            Record("Colors") {
+                Field("Red", Type.FromString "int")
+                EscapeHatch(customField)
```

Comment:
Good point to unit test the use of EscapeHatch.
You may also want to add a unit test to yield directly the `customField` value without using `EscapeHatch`.

I see that you already have the corresponding yield:
https://github.com/edgarfgp/Fabulous.AST/pull/20/files#diff-e6091de70117e4029a157940d62c59db3043a3cdd505c8c6e9750d6137a994d9R73-R76

Reply:
I'm actually not a fan of the direct yield, to be honest.
I'd be more in favour of removing that corresponding yield.
It makes it more obvious that the user is indeed using the escape hatch.

Reply:
I can see the value of both approaches here. I will think about this and maybe remove it before v 1.0